webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx

```diff
@@ -50,23 +54,38 @@ const ExperimentCheckpoints: React.FC<Props> = ({ experiment, pageRef }: Props)
 
   const { settings, updateSettings } = useSettings<Settings>(settingsConfig);
 
+  const handleOnCloseCreateModel = useCallback((
+    reason?: ModalCloseReason,
+    checkpoints?: string[],
+    modelName?: string,
+  ) => {
+    if (checkpoints) openModalCheckpointRegister({ checkpoints, selectedModelName: modelName });
+  }, [ openModalCheckpointRegister ]);
+
   const {
     contextHolder: modalModelCreateContextHolder,
     modalOpen: openModalCreateModel,
-  } = useModalModelCreate();
+  } = useModalModelCreate({ onClose: handleOnCloseCreateModel });
 
   const handleOnCloseCheckpointRegister = useCallback((
     reason?: ModalCloseReason,
-    checkpointUuid?: string,
+    checkpoints?: string[],
   ) => {
-    if (checkpointUuid) openModalCreateModel({ checkpointUuid });
+    if (checkpoints) openModalCreateModel({ checkpoints });
   }, [ openModalCreateModel ]);
 
-  const {
+  // Has to use var to hoist openModalCheckpointRegister for use above
```

Comment:
Question: it looks like we are able to avoid doing this in CheckpointModalTrigger. Would it be possible to do same here?

Alternatively, should useModalCheckpointRegister wrap the useModalModelCreate?